compiler/src/iree/compiler/Codegen/Common/FoldTensorSubsetIntoVectorTransferOps.cpp

```diff
@@ -274,15 +274,9 @@ class FoldExtractSliceIntoTransferWrite final
          llvm::zip_equal(llvm::seq<int64_t>(0, inBounds.size()),
                          xferOp.getVectorType().getShape(), mixedSizes)) {
       auto maybeCst = getConstantIntValue(destSize);
-      if (!maybeCst) {
-        inBounds[idx] = false;
+      inBounds[idx] = false;
```

Comment:
@hanhanW Made changes to here to make `in_bounds` attribute conformant. 

Before:
```
%7 = vector.transfer_write %5, %6[%c0, %c0] {in_bounds = [true, true]} : vector<8x2xi32>, tensor<3x2xi32>
```
After:
```
%7 = vector.transfer_write %5, %6[%c0, %c0] {in_bounds = [false, true]} : vector<8x2xi32>, tensor<3x2xi32>
```

Reply:
Changes to this file should go to their own separate PR, which should include a corresponding change to a lit test (if no lit test existed for this behaviour, one should be added) and merged first.

The current change sets `in_bounds = false` unconditionally. Up to @hanhanW , but I would expect that we would like to still generate `in_bounds = true` in cases where this can be reasonably easy proven to be correct. That is, when the offset/sizes can be determined at compile time to be in bounds.

Reply:
Test added. Now need @hanhanW 's input about the `in_bounds` attribute value.

Reply:
Meanwhile you can start splitting this into a separate PR. Until @hanhanW weighs in, default to assuming that it is desirable to still generate `in_bounds = true` when possible.

Reply:
> Changes to this file should go to their own separate PR, which should include a corresponding change to a lit test (if no lit test existed for this behaviour, one should be added) and merged first.

+1

> I would expect that we would like to still generate in_bounds = true in cases where this can be reasonably easy proven to be correct. That is, when the offset/sizes can be determined at compile time to be in bounds.

+1 on what Benoit said.